qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -101,7 +101,7 @@ class OptimizationResultStatus(Enum):
 
 
 class OptimizationResult:
-    """The optimization result class.
+    """An abstract class for optimization results.
```

Comment:
This is not an Abstract class though - unlike the OptimizationAlgorithm. In fact some, like cplex_optimizer return this directly as their result since its a concrete class that can be instantiated https://github.com/Qiskit/qiskit-aqua/blob/e13728e6cdfdb1438a7d2d11d856622c90ce328b/qiskit/optimization/algorithms/cplex_optimizer.py#L105

Reply:
You are right. How about `A base class  for optimization results'?